graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/CustomGraphQLClient.kt

```diff
@@ -21,7 +21,7 @@ package com.netflix.graphql.dgs.client
  * The user is responsible for doing the actual HTTP request, making this pluggable with any HTTP client.
  * For a more convenient option, use [WebClientGraphQLClient] instead.
  */
-class CustomGraphQLClient(private val url: String, private val requestExecutor: RequestExecutor) : GraphQLClient {
+class CustomGraphQLClient(private val url: String, private val requestExecutor: RequestExecutor, private val customDeserializer: Map<Class<*>, (Any) -> Any?>? = null) : GraphQLClient {
```

Comment:
This needs to be defined as a separate constructor, otherwise it will be a required argument for Java users.